src/definitions.jl

```diff
@@ -415,6 +415,11 @@ Base.step(x::Frequencies) = x.multiplier
 
 Base.copy(x::Frequencies) = x
 
+# Retain the lazy representation upon scalar multiplication
+Broadcast.broadcasted(::typeof(*), f::Frequencies, x::Number) = Frequencies(f.n_nonnegative, f.n, f.multiplier * x)
+Broadcast.broadcasted(::typeof(*), x::Number, f::Frequencies) = Broadcast.broadcasted(*, f, x)
+Broadcast.broadcasted(::typeof(/), f::Frequencies, x::Number) = Frequencies(f.n_nonnegative, f.n, f.multiplier / x)
```

Comment:
I guess for completeness we should also have
```jl
Broadcast.broadcasted(::typeof(\), x::Number, f::Frequencies) = Broadcast.broadcasted(/, f, x)
```

Reply:
I've added this now